gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -301,15 +305,17 @@ public static boolean isWithinRange(String cronExpression, int maxNumDaysToSched
       if (nextValidTimeAfter == null) {
         log.warn("Calculation issue for next valid time for expression: {}. Will default to true for within range",
             cronExpression);
-        return true;
+        return false;
```

Comment:
please characterize why the change from current/prior behavior--was it a faulty presumption?

a code comment would go a long way in communicating the reasoning.

aso, log msg is now out-of-sync w/ behavior

Reply:
I have added the comments clarifying the need for change in my latest commit.